src/crypto/tls/handshake_client_tls13.go

```diff
@@ -573,14 +659,24 @@ func (hs *clientHandshakeStateTLS13) sendClientCertificate() error {
 	certVerifyMsg := new(certificateVerifyMsg)
 	certVerifyMsg.hasSignatureAlgorithm = true
 
-	certVerifyMsg.signatureAlgorithm, err = selectSignatureScheme(c.vers, cert, hs.certReq.supportedSignatureAlgorithms)
+	sigAlgorithm, err := selectSignatureScheme(c.vers, cert, hs.certReq.supportedSignatureAlgorithms)
 	if err != nil {
 		// getClientCertificate returned a certificate incompatible with the
 		// CertificateRequestInfo supported signature algorithms.
 		c.sendAlert(alertHandshakeFailure)
 		return err
 	}
 
+	if certMsg.delegatedCredential {
+		dCred, err := unmarshalDelegatedCredential(certMsg.certificate.DelegatedCredential)
+		if err != nil {
+			return errors.New("tls: failed: " + err.Error())
+		}
+		certVerifyMsg.signatureAlgorithm = dCred.cred.expCertVerfAlgo
```

Comment:
should this not also be done via `selectSignatureAlgorithm`?

Reply:
if `expCertVerifyAlgo` is verified against `supportedSignatureAlgorithmsDC` elsewhere, maybe a comment along those lines would be a good idea.

Reply:
It is changed now, as it was wrong ;)